### PR TITLE
fixed failure to check item deletion

### PR DESCRIPTION
**Pull request by nrhine1**
forgot to add checking of item deletion of a read-only SqliteDict in the last PR... sorry about that.

### CODE REVIEW

**Comment by tmylk** [comment]
NP. Do clear() and update()  need checks too?

**Comment by tmylk** [comment]
And terminate()?

**Comment by nrhine1** [comment]
Yes. Will fix

On Mon, Oct 12, 2015 at 10:48 AM, Lev Konstantinovskiy <
notifications@github.com> wrote:

> And terminate()?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/piskvorky/sqlitedict/pull/38#issuecomment-147420453>.
>



-- 
Nick Rhinehart <http://www.nickrhinehart.com>
PhD student
Robotics Institute
Carnegie Mellon University

**Comment by nrhine1** [comment]
whoops, one sec

File: sqlitedict.py [hunk 1]
```diff
@@ -287,6 +296,9 @@ def close(self, do_log=True):
                 pass
 
     def terminate(self):
+        if self.flag == 'r':
+            raise RuntimeError('Refusing to terminate read-only SqliteDict')
+
         """Delete the underlying database file. Use with care."""
```
Comment (piskvorky) [inline sqlitedict.py]: We don't want code above the docstring.

**Comment by piskvorky** [comment]
Merging. Thanks @nrhine1 !

**Pull request merged**